pkg/kubelet/cm/topologymanager/policy_options_test.go

```diff
@@ -39,55 +39,67 @@ type optionAvailTest struct {
 
 func TestNewTopologyManagerOptions(t *testing.T) {
 	testCases := []struct {
-		description     string
-		policyOptions   map[string]string
-		featureGate     featuregate.Feature
-		expectedErr     error
-		expectedOptions PolicyOptions
+		description       string
+		policyOptions     map[string]string
+		featureGate       featuregate.Feature
+		featureGateEnable bool
+		expectedErr       error
+		expectedOptions   PolicyOptions
 	}{
 		{
-			description: "return TopologyManagerOptions with PreferClosestNUMA set to true",
-			featureGate: pkgfeatures.TopologyManagerPolicyAlphaOptions,
+			description:       "return TopologyManagerOptions with PreferClosestNUMA set to true",
+			featureGate:       pkgfeatures.TopologyManagerPolicyBetaOptions,
+			featureGateEnable: true,
 			expectedOptions: PolicyOptions{
 				PreferClosestNUMA: true,
 			},
 			policyOptions: map[string]string{
 				PreferClosestNUMANodes: "true",
 			},
 		},
+		{
+			description: "fail to set option when TopologyManagerPolicyBetaOptions feature gate is not set",
+			featureGate: pkgfeatures.TopologyManagerPolicyBetaOptions,
+			policyOptions: map[string]string{
+				PreferClosestNUMANodes: "true",
+			},
+			expectedErr: fmt.Errorf("Topology Manager Policy Beta-level Options not enabled,"),
+		},
 		{
 			description: "return empty TopologyManagerOptions",
 		},
 		{
-			description: "fail to parse options",
-			featureGate: pkgfeatures.TopologyManagerPolicyAlphaOptions,
+			description:       "fail to parse options",
+			featureGate:       pkgfeatures.TopologyManagerPolicyAlphaOptions,
+			featureGateEnable: true,
 			policyOptions: map[string]string{
 				PreferClosestNUMANodes: "not a boolean",
 			},
 			expectedErr: fmt.Errorf("bad value for option"),
 		},
 		{
-			description: "test beta options success",
-			featureGate: pkgfeatures.TopologyManagerPolicyBetaOptions,
+			description:       "test alpha options success",
+			featureGate:       pkgfeatures.TopologyManagerPolicyAlphaOptions,
+			featureGateEnable: true,
 			policyOptions: map[string]string{
-				fancyBetaOption: "true",
+				fancyAlphaOption: "true",
 			},
 		},
 		{
-			description: "test beta options success",
+			description: "test alpha options fail",
 			policyOptions: map[string]string{
-				fancyBetaOption: "true",
+				fancyAlphaOption: "true",
 			},
-			expectedErr: fmt.Errorf("Topology Manager Policy Beta-level Options not enabled,"),
+			expectedErr: fmt.Errorf("Topology Manager Policy Alpha-level Options not enabled,"),
```

Comment:
this looks OK, but do we want to keep the old test? (not replace it)

Reply:
We have previous test but it had to be changed a bit to accomodate that not beta feature gate is enabled:
```go
	{
			description: "fail to set option when TopologyManagerPolicyBetaOptions feature gate is not set",
			featureGate: pkgfeatures.TopologyManagerPolicyBetaOptions,
			policyOptions: map[string]string{
				PreferClosestNUMANodes: "true",
			},
			expectedErr: fmt.Errorf("Topology Manager Policy Beta-level Options not enabled,"),
		},
```

Reply:
added old test